libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/AkkaTracing.scala

```diff
@@ -0,0 +1,99 @@
+package com.advancedtelematic.libtuf_server
+
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.http.scaladsl.server.{Directive0, Directive1}
+import brave.{Span, Tracing}
+import brave.http.{HttpClientAdapter, HttpServerAdapter, HttpServerHandler, HttpTracing}
+import brave.propagation.{Propagation, TraceContext}
+import zipkin2.reporter.AsyncReporter
+import zipkin2.reporter.okhttp3.OkHttpSender
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.duration.span
+
+object AkkaTracing {
+
+  class RequestTracing(httpTracing: HttpTracing, requestSpan: Span) {
```

Comment:
I was trying to encapsulate all the state here, I think we can make this much better than this but the client library is a bit.... complicated.

Reply:
_Complicated_ 😆